error-handling/src/test/java/com/palantir/remoting1/errors/SerializableErrorErrorDecoderTests.java

```diff
@@ -54,74 +54,75 @@ public void testNotAuthorizedException() throws Exception {
         NotAuthorizedException originalException =
                 new NotAuthorizedException(message, Response.status(Status.UNAUTHORIZED).build());
 
-        Exception exception = encodeAndDecode(originalException);
+        RemoteException exception = (RemoteException) encodeAndDecode(originalException);
+        assertThat(exception.getCause(), is(nullValue()));
+        assertThat(exception.getStatus(), is(Status.UNAUTHORIZED.getStatusCode()));
+        assertThat(exception.getRemoteException().getExceptionName(), is(NotAuthorizedException.class.getName()));
         assertThat(exception.getMessage(), is(message));
-        assertEquals(Status.UNAUTHORIZED, ((WebApplicationException) exception).getResponse().getStatusInfo());
-        assertThat(exception.getCause().getMessage(), is(message));
-        assertEquals(Status.UNAUTHORIZED,
-                ((WebApplicationException) exception.getCause()).getResponse().getStatusInfo());
     }
 
     @Test
     public void testSpecificException() {
         Exception exception = encodeAndDecode(new IllegalArgumentException("msg"));
-        assertThat(exception, is(instanceOf(IllegalArgumentException.class)));
+        assertThat(exception, is(instanceOf(RemoteException.class)));
         assertThat(exception.getMessage(), is("msg"));
     }
 
     @Test
     public void testTextPlainException() {
-        Exception decode = decode(MediaType.TEXT_PLAIN, "errorbody");
+        Exception decode = decode(MediaType.TEXT_PLAIN, 42, "errorbody");
```

Comment:
done

On Mon, Aug 15, 2016 at 7:32 AM Joe Ellis <notifications@github.com> wrote:

> In
> error-handling/src/test/java/com/palantir/remoting1/errors/SerializableErrorErrorDecoderTests.java
> <https://github.com/palantir/http-remoting/pull/199#discussion_r74770006>:
>
> >          assertThat(exception.getMessage(), is("msg"));
> >      }
> >
> >      @Test
> >      public void testTextPlainException() {
> > -        Exception decode = decode(MediaType.TEXT_PLAIN, "errorbody");
> > +        Exception decode = decode(MediaType.TEXT_PLAIN, 42, "errorbody");
>
> wanna make it a constant?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/http-remoting/pull/199/files/bc2affd0f624fb975b783154957f8d3321ebc2c6#r74770006>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGOdwWHnIaZqris4WqdVvSO63vOS8AT-ks5qgHhVgaJpZM4Jjywx>
> .
>